File: oauth2-server/src/main/java/a/gleb/keycloaktojwt/db/repository/AccountRepository.java
package a.gleb.keycloaktojwt.db.repository;

import a.gleb.keycloaktojwt.db.entity.Account;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.lang.NonNull;

import java.util.Optional;
import java.util.UUID;

public interface AccountRepository extends JpaRepository<Account, UUID> {

    Optional<Account> findAccountByUsername(@NonNull String username);
}

File: oauth2-server/src/main/kotlin/a/gleb/oauth2server/db/repository/AccountRepository.kt
package a.gleb.oauth2server.db.repository

import a.gleb.oauth2server.db.entity.Account
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface AccountRepository: JpaRepository<Account, UUID> {

    fun findAccountByUsername(username: String): Account?
}